### PR TITLE
Add container hostname info

### PR DESCRIPTION
**Pull request by bdevcich-hpe**

### CODE REVIEW

File: docs/guides/user-containers/readme.md [hunk 1]
```diff
@@ -6,7 +6,55 @@ applications to be run on Rabbit nodes with access to NNF ephemeral and persiste
...
+### Container Environment Variables
+
+These variables are provided for use inside the container. They can be used as part of the
+container command in the NNF Container Profile or within the container itself.
+
+#### Container Hostnames and Domains
+
+Containers can contact one another via Kubernetes cluster networking. This functionality is provided by DNS. Environment variables are provided that allow a user to be able to piece together the FQDN so that the other containers can be contacted.
+
+This example demonstrates an MPI container workflow, with two worker nodes. Two worker nodes means two containers running on two NNF nodes.
```
Comment (ajfloeder) [inline docs/guides/user-containers/readme.md:24]: "Two worker nodes means two containers running on to NNF nodes."
So these are generic k8s worker nodes, explicitly not Rabbits? Or are they something else?

File: docs/guides/user-containers/readme.md [hunk 2]
```diff
@@ -6,7 +6,55 @@ applications to be run on Rabbit nodes with access to NNF ephemeral and persiste
...
+
+```console
+mpiuser@my-container-workflow-launcher:~$ cat /etc/mpi/hostfile
+my-container-workflow-worker-0.my-container-workflow-worker.default.svc slots=1
+my-container-workflow-worker-1.my-container-workflow-worker.default.svc slots=1
+```
+
+### Compute Node Environment Variables
+
+These environment variables are provided to the compute node via Flux by way of the DWS Workflow. Note that these environment variables are consistent across all the compute nodes for a given workflow. At this time, there is no way to provided compute specific environment variables.
```
Comment (ajfloeder) [inline docs/guides/user-containers/readme.md:45]: Do you ever intend to provide compute specific environment variables? If not or at least not likely, consider removing "At this time,"

Comment (roehrich-hpe) [inline docs/guides/user-containers/readme.md:45]: provided => provide

File: docs/guides/user-containers/readme.md [hunk 3]
```diff
@@ -6,7 +6,55 @@ applications to be run on Rabbit nodes with access to NNF ephemeral and persiste
...
+
+#### `NNF_CONTAINER_PORTS`
+
+If container ports are requested via the NNF Container Profile, then this environment variable will provide the allocated ports for the container. This is a comma separated list of ports if multiple ports are requested.
+
+This allows an application on the compute node to contact the user container running on its local NNF node via these port numbers. The compute nodes must have proper routing to the NNF Node and needs a generic way of contacting the NNF node. It is suggested than a DNS entry is provided via `/etc/hosts`, or similar.
```
Comment (roehrich-hpe) [inline docs/guides/user-containers/readme.md:51]: The compute nodes => the compute node

File: docs/guides/user-containers/readme.md [hunk 4]
```diff
@@ -6,7 +6,55 @@ applications to be run on Rabbit nodes with access to NNF ephemeral and persiste
...
+my-container-workflow-worker-1.my-container-workflow-worker.default.svc slots=1
+```
+
+### Compute Node Environment Variables
+
+These environment variables are provided to the compute node via Flux by way of the DWS Workflow. Note that these environment variables are consistent across all the compute nodes for a given workflow. At this time, there is no way to provided compute specific environment variables.
+
+#### `NNF_CONTAINER_PORTS`
+
+If container ports are requested via the NNF Container Profile, then this environment variable will provide the allocated ports for the container. This is a comma separated list of ports if multiple ports are requested.
```
Comment (ajfloeder) [inline docs/guides/user-containers/readme.md:49]:
```suggestion
If the NNF Container Profile requests container ports, then this environment variable provides the allocated ports for the container. This is a comma separated list of ports if multiple ports are requested.
```

File: docs/guides/user-containers/readme.md [hunk 5]
```diff
@@ -99,11 +147,11 @@ Hub username and the email address they have associated with that username.  In
 this case, the secret will be named `readonly-red-rock-slushy`.
 
 ```console
-$ USER_TOKEN=users-token-text
-$ USER_NAME=dean
-$ USER_EMAIL=dean@myco.com
-$ SECRET_NAME=readonly-red-rock-slushy
-$ kubectl create secret docker-registry $SECRET_NAME -n default --docker-server="https://index.docker.io/v1/" --docker-username=$USER_NAME --docker-password=$USER_TOKEN --docker-email=$USER_EMAIL
+USER_TOKEN=users-token-text
```
Comment (roehrich-hpe) [inline docs/guides/user-containers/readme.md:150]: I tend to get disoriented without the prompt, but I guess this is a regular area where heads bump.

**Review by roehrich-hpe — APPROVED** [review]

File: docs/guides/user-containers/readme.md (hunk 3 above)
Comment (ajfloeder) [inline docs/guides/user-containers/readme.md:51]:
```suggestion
This allows an application on the compute node to contact the user container running on its local NNF node via these port numbers. The compute node must have proper routing to the NNF Node and needs a generic way of contacting the NNF node. It is suggested than a DNS entry is provided via `/etc/hosts`, or similar.
```

**Review by ajfloeder — APPROVED** [review]

File: docs/guides/user-containers/readme.md (hunk 1 above)
Comment (bdevcich-hpe) [reply docs/guides/user-containers/readme.md:24]: This should say two worker pods, not nodes.

File: docs/guides/user-containers/readme.md (hunk 2 above)
Comment (bdevcich-hpe) [reply docs/guides/user-containers/readme.md:45]: I don't believe Flux has a way to do this, unless we get specific with the envionrment variable names themselves. I'll remove that part.

File: docs/guides/user-containers/readme.md (hunk 5 above)
Comment (bdevcich-hpe) [reply docs/guides/user-containers/readme.md:150]: I use them only when there is output. My markdown linter auto-corrected it: https://github.com/DavidAnson/markdownlint/blob/v0.29.0/doc/md014.md.

File: docs/guides/user-containers/readme.md [hunk 6]
```diff
@@ -6,7 +6,55 @@ applications to be run on Rabbit nodes with access to NNF ephemeral and persiste
 !!! note
 
     The following is a limited look at User Containers.  More content will be
-    provided after the RFC has been finalized.
+    provided after the RFC has been finalized. This is a work in progress.
+
+## Environment Variables
+
+Two sets of environment variables are available with container workflows: Container and Compute Node. The former are the variables that are available inside the user containers. The latter are the variables that are provided back to the DWS workflow, which in turn are collected by Flux and provided to compute nodes. See the Flux documentation for more details.
```
Comment (matthew-richerson) [inline docs/guides/user-containers/readme.md:13]: I try to avoid calling out Flux, and just say "WLM" instead. Maybe over cautious.

**Review by matthew-richerson — APPROVED** [review]

File: docs/guides/user-containers/readme.md (hunk 6 above)
Comment (bdevcich-hpe) [reply docs/guides/user-containers/readme.md]: Removed

**Pull request merged**